proposals/NNNN-support-for-android-custom-routerservice.md

```diff
@@ -0,0 +1,275 @@
+# Support for Android custom RouterService
+* Proposal: [SDL-NNNN](NNNN-support-for-android-custom-routerservice.md)
+* Author: [Shinichi Watanabe](https://github.com/shiniwat)
+* Status: **Awaiting review**
+* Impacted Platforms: Android
+
+## Introduction
+This proposal is to improve the case where SDL application needs to work with custom RouterService. The definition of custom RouterService is varied, but this proposal refers to the case where a head unit requires an OEM-specific app that works as a RouterService.
+For instance, if an OEM uses specific Bluetooth server UUID that differs from SDL standard's one, that should be custom RouterService.
+
+## Motivation
+
+Currently, AndroidManifest.xml can indicate the application has custom RouterService, something like below:
+```java
+    <service
+        android:name=".SdlRouterService"
+        android:enabled="true"
+        android:exported="true"
+        android:process="com.smartdevicelink.router">
+        <intent-filter>
+            <action android:name="com.smartdevicelink.router.service" />
+        </intent-filter>
+        <meta-data android:name="sdl_custom_router" android:value="true" />
+        <meta-data android:name="sdl_router_version" android:value="@integer/sdl_router_service_version_value" />
+    </service>
+```
+
+When SDL application determines which RouterService to bind with, the app creates the list of SDL enabled apps, and checks to see if a RouterService whose transport connects with a head unit.
+If a custom RouterService is already connected with a head unit, and a SDL application can identify the RouterService correctly, the app should work fine. However, if not, the application instantiates "possibly the best" RouterService. In this case, custom RouterService is less likely chosen, because custom RouterService has lower priority than non-custom RouterService.
```

Comment:
Thanks for the feedback. I have slightly modified the sentense to clarify the if-then logic in this sentense:

If a custom RouterService is already connected with a head unit, and if a SDL application can identify the RouterService correctly, then the app should work fine. However, if not, the application instantiates "possibly the best" RouterService. In this case, the custom RouterService is less likely chosen, because the custom RouterService has lower priority than the non-custom RouterService.